### ISSUE TITLE
CA1508 false positive in generic nullable

### ISSUE THREAD

**Issue by Cyl18** [dotnet/roslyn-analyzers#4795]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code (code analysis)`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug + Reproduce
I have enabled all analyzers by `<AnalysisMode>AllEnabledByDefault</AnalysisMode>`

```csharp
    public sealed class ImmutableNullableContainer<T>
    {
        [MaybeNull]
        public T? Value { [return: MaybeNull] get; }

        public ImmutableNullableContainer(T? value)
        {
            Value = value;
        }

        public override string? ToString()
            => Value.ToString();
    }
```

![image](https://user-images.githubusercontent.com/14993992/107114815-c476a580-68a3-11eb-9400-38ecd856797c.png)
it shows CS8602.
if I add a `?` after `Value`, it shows CA1508:
![image](https://user-images.githubusercontent.com/14993992/107114845-f5ef7100-68a3-11eb-92e7-ab80bfdd9461.png)

more discussions: https://github.com/dotnet/csharplang/discussions/4399

> *I'm still learning English, please excuse my typos, and you can point them out :)

**Comment by Youssef1313** [comment]
This looks a false positive, not a false negative. If so, can you update the issue title?

**Comment by Youssef1313** [comment]
@mavasani Have your PR https://github.com/dotnet/roslyn-analyzers/pull/4554 fixed this?

**Comment by mavasani** [comment]
This no longer repros on the latest .NET 8 preview SDK
